app/javascript/components/visualization/ScatterPlot.jsx

```diff
@@ -355,34 +372,69 @@ function RawScatterPlot({
   useEffect(() => {
     setIsLoading(true)
 
+    let expressionArray
+
+    const fetchMethod = dataCache ? dataCache.fetchCluster : fetchCluster
+
     // use an image and/or data cache if one has been provided, otherwise query scp-api directly
     if (
       flags?.progressive_loading && isGeneExpression(genes, isCorrelatedScatter) && !isAnnotatedScatter &&
-      !scatterData &&
-      genes[0] === 'A1BG-AS1' // Placeholder; likely replace with setting like DE
+      !scatterData
     ) {
-      const bucketName = 'broad-singlecellportal-public'
-      const filePath = `test/scatter_image/${genes[0]}-v2.webp`
-      fetchBucketFile(bucketName, filePath).then(async response => {
-        renderImage(response)
+      const urlSafeCluster = cluster.replaceAll(' ', '_')
+      const gene = genes[0]
+      const stem = '_scp_internal/cache/expression_scatter/'
+      const leaf = `${urlSafeCluster}/${gene}`
+
+      // TODO (SCP-4839): Instrument more bucket cache analytics, then remove line below
+      // window.t0 = Date.now()
+
+      const imagePath = `${stem}images/${leaf}.webp`
+      const dataPath = `${stem}data/${leaf}.json`
+
+      const expressionParams = {
+        studyAccession,
+        cluster,
+        annotation: annotation ? annotation : '',
+        subsample,
+        consensus,
+        genes,
+        isAnnotatedScatter,
+        isCorrelatedScatter
+      }
+
+      fetchBucketFile(bucketId, imagePath).then(async response => {
+        const imageCacheHit = response.ok
+
+        // Draw plot as static image first, if it's cached
+        if (imageCacheHit) {
+          await drawPlotImage(response)
+        }
+
+        // Then make it interactive, using gene expression scatter plot data array from GCS bucket
+        renderBucketData(fetchMethod, expressionParams, dataPath)
```

Comment:
Non-blocking: since this is still WIP and prototype-level, I don't consider this blocking because it's an error state that I doubt we'd encounter in normal usage.

I think this method should have error handling such that if we make a `GET` for the gene-level JSON data and can't find it, it will then fall back to the API.  While testing this PR, the image request returned `200` but the data request was `404` because it wasn't in the bucket.  I think the testing setup missed steps for copying that data or relied on the state from a previous PR review, and I had done cleanup while testing other work.

Reply:
Agreed, this seems very unlikely on production.  I've updated the step 5 above to account for that test data case, and opened SCP-4845 for the related robustness task.